qiskit/providers/v2/options.py

```diff
@@ -0,0 +1,36 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Container class for backend options."""
+
+
+class Options:
+    """Base options object
+
+    This class is the abstract class that all backend options are based
+    on. The properties of the class are intended to be all ddynamically
+    adjustable so that a user can reconfigure the backend on demand. If a
+    property is immutable to the user (eg something like number of qubits)
+    that should be a property of the backend class (itself instead of the
+    options.
+    """
+
+    def __init__(self, **kwargs):
+        self.data = kwargs if kwargs else {}
+
+    def update_options(self, **fields):
```

Comment:
is it possible to make them into attributes of `Options` so updating them looks like `backend.options.shots=2000`?

Reply:
Yeah, @jyu00's suggestion above about using `SimpleNamespace` should enable that pattern for setting it. I'd still like to keep the method because it has the membership check with a custom exception.

Reply:
I think you still need a `@options.setter` in `Backend` to do `backend.options.shots`, since right now `_options` is the real attribute. Maybe just convert `set_options()` to a setter?

Reply:
I think it should be good in it's current form because there is a `@parameter` decorator on `options`. So `backend.options` will return the `Options` object and then `.shots =` will call setattr on that object. If we added a setter on options to `Backend` it would be called when `Backend.options =`. I do think I should try to figure out if there is a way to add a `__setattr__` method to this class to call `update_options` since right now you could call `Backend.options.secret_shots = 10` and that would get set even though its probably not a valid option. But, maybe that's not a problem